interpreter/src/ser.rs

```diff
@@ -568,8 +568,32 @@ mod tests {
     use serde_bytes::Bytes;
     use std::{collections::HashMap, iter::FromIterator, sync::Arc};
 
+    macro_rules! primitive_test {
+        ($functionName:ident, $strValue: literal, $value: expr) => {
+            #[test]
+            fn $functionName() {
+                let program = Program::compile($strValue).unwrap();
+                let result = program.execute(&Context::default());
+                assert_eq!(Value::from($value), result.unwrap());
+            }
+        };
+    }
+
+    primitive_test!(test_u64_zero, "0u", 0_u64);
+    primitive_test!(test_i64_zero, "0", 0_i64);
+    primitive_test!(test_f64_zero, "0.0", 0_f64);
+    //primitive_test!(test_f64_zero, "0.", 0_f64); this test fails
+    primitive_test!(test_bool_false, "false", false);
+    primitive_test!(test_bool_true, "true", true);
+    primitive_test!(test_string_empty, "\"\"", "");
+    primitive_test!(test_string_non_empty, "\"test\"", "test");
+    primitive_test!(test_byte_ones, r#"b"\001\001""#, vec!(1_u8, 1_u8));
+    // primitive_test!(test_triple_double_quoted_string, #"r"""""""#, "");
+    // primitive_test!(test_triple_single_quoted_string, "r''''''", "");
```

Comment:
Can these be removed if they're not going to be used?

Reply:
These will be used, however they currently fail. I can remove them until they pass if you prefer.

Reply:
Normally my preference is to either (1) get the test working or (2) exclude the test entirely, but it sounds like a fix for these tests is on the way as well so I'm good to merge now.